swap/peer.go

```diff
@@ -36,10 +36,11 @@ var ErrDontOwe = errors.New("no negative balance")
 // Peer is a devp2p peer for the Swap protocol
 type Peer struct {
 	*protocols.Peer
-	swap            *Swap
-	backend         cswap.Backend
-	beneficiary     common.Address
-	contractAddress common.Address
+	swap               *Swap
+	backend            cswap.Backend
+	beneficiary        common.Address
+	contractAddress    common.Address
+	lastReceivedCheque *Cheque
```

Comment:
This is the nicer approach but I wonder, shouldn't we have a consistent handling if you introduce this? Shouldn't we also have `lastSentCheque` then and have their correspondent methods too?

Reply:
Yes, that's the idea (move everything that only relates to one peer from swap to peer). But I think that refactor should be done in a separate PR.

Reply:
I am not sure I agree, because this  PR is introducing a change which makes a situation inconsistent which was, even if maybe a bit ugly, consistent before. So I would have nothing against refactoring it inside this same PR.

Nevertheless to speed up things let's move on, I created an issue for this: https://github.com/ethersphere/swarm/issues/1604, but please let's take that issue soon.

Reply:
I can start working on it as soon as this PR is merged but it's not gonna be ready by the time we want to merge to master.

Reply:
yes agreed. It will be in after the merge to `master` - unless reviewers of the `master` PR object for some reason